File: arisan/src/main/java/com/github/arisan/helper/FieldUtils.java
package com.github.arisan.helper;

import com.github.arisan.model.FormModel;
import com.github.arisan.model.FormModel;
import com.github.arisan.model.RadioModel;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.util.ArrayList;
import java.util.List;

public class FieldUtils {
    public static List<String> countBlank(List<FormModel> models){
        List<String> blank_field = new ArrayList<>();
        for(FormModel model:models){
            if(model.getValue()==null && model.isRequire()){
                blank_field.add(model.getLabel());
            }else if(model.getChildFieldModel()!=null){
                for(List<FormModel> childList:model.getChildFieldModel()){
                    for(FormModel child:childList){
                        if(child.getValue()==null && child.isRequire()){
                            blank_field.add(child.getLabel()+"["+model.getChildFieldModel().indexOf(childList)+"]");
                        }
                    }
                }
            }
        }
        return blank_field;
    }

    public static List<String> countFormBlank(List<FormModel> models){
        List<String> blank_field = new ArrayList<>();
        for(FormModel model:models){
            if(model.getValue()==null && model.isRequire()){
                blank_field.add(model.getLabel());
            }else if(model.getChildFieldModel()!=null){
                for(List<FormModel> childList:model.getChildFieldModel()){
                    for(FormModel child:childList){
                        if(child.getValue()==null && child.isRequire()){
                            blank_field.add(child.getLabel()+"["+model.getChildFieldModel().indexOf(childList)+"]");
                        }
                    }
                }
            }
        }
        return blank_field;
    }

    public static List<FormModel> removeField(String fieldName, List<FormModel> list){
        for(FormModel model:list){
            if (model.getName().equals(fieldName)){
                list.remove(model);
                break;
            }
        }
        return list;
    }

    public static List<FormModel> insertOrUpdateField(FormModel new_model, List<FormModel> list){
        removeField(new_model.getName(),list);
        list.add(new_model);
        return list;
    }

    public static List<String> convertArrayToList(Object data){
        Gson gson = new Gson();
        String json = gson.toJson(data);
        return gson.fromJson(json,new TypeToken<List<String>>(){}.getType());
    }

    public static Object[] convertListToArray(Object data){
        Gson gson = new Gson();
        String json = gson.toJson(data);
        return gson.fromJson(json,Object[].class);
    }

    public static List<RadioModel> convertDataToRadio(Object data){
        Gson gson = new Gson();
        String json = gson.toJson(data);
        List<String> list = gson.fromJson(json,new TypeToken<List<String>>(){}.getType());
        List<RadioModel> radioModels = new ArrayList<>();
        for(String s : list){
            RadioModel rm = new RadioModel();
            rm.setId(list.indexOf(s));
            rm.setValue(s);
            radioModels.add(rm);
        }
        return radioModels;
    }
}

File: arisan/src/main/java/com/github/arisan/helper/FieldUtils.kt
package com.github.arisan.helper

import com.github.arisan.model.ArisanFieldModel
import com.github.arisan.model.RadioModel
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

object FieldUtils {
    fun countBlank(models:MutableList<ArisanFieldModel>?): List<String>? {
        return models?.filter { it.isRequire && it.value == null }?.map { it.label }
    }

    fun removeField(fieldName: String, list: MutableList<ArisanFieldModel>): List<ArisanFieldModel> {
        return list.filter { it.name != fieldName }
    }

    fun insertOrUpdateField(new_model: ArisanFieldModel, list: MutableList<ArisanFieldModel>): List<ArisanFieldModel> {
        removeField(new_model.name, list)
        list.add(new_model)
        return list
    }

    fun convertArrayToList(data: Any?): List<String>? {
        val gson = Gson()
        val json = gson.toJson(data)
        return gson.fromJson<List<String>>(json, object : TypeToken<List<String>>() {

        }.type)
    }

    fun convertDataToRadio(data: Any): List<RadioModel>? {
        val gson = Gson()
        val json = gson.toJson(data)
        return gson.fromJson<List<RadioModel>>(json, object : TypeToken<List<RadioModel>>() {

        }.type)
    }
}
